system/jlib/jio.ipp

```diff
@@ -99,6 +99,80 @@ protected:
         curBufferOffset += sizeGot;
         return sizeGot;
     }
+    inline bool readLineFromBuffer(bool keepCRLF, StringBuffer & outBuffer) 
+    {
+        if ((numInBuffer == curBufferOffset) && !fillBuffer())
+            return true;
+
+        bool eof = false;
+        bool foundCRLF = false;
+        while (true)
+        {
+            size32_t outLength = 0;
+            char * outStartFrom = (char *) buffer+curBufferOffset;
+            size32_t tempOffset = curBufferOffset;
+            while (tempOffset < numInBuffer)
+            {
+                if (buffer[tempOffset] == '\r') // standard case, \r\n marks a new header line.
```

Comment:
minor: would probably be a bit more efficient to use a pointer to the buffer that you increment here.

Reply:
Will change to a pointer.